Tests/SwiftDocCTests/Model/RenderContentMetadataTests.swift

```diff
@@ -297,4 +297,26 @@ class RenderContentMetadataTests: XCTestCase {
             default: XCTFail("Unexpected element")
         }
     }
+    
+    func testHeadingAnchorShouldBeEncoded() throws {
+        let (bundle, context) = try testBundleAndContext(named: "TestBundle")
+        var renderContentCompiler = RenderContentCompiler(context: context, bundle: bundle, identifier: ResolvedTopicReference(bundleIdentifier: bundle.identifier, path: "/path", fragment: nil, sourceLanguage: .swift))
+        
+        let source = """
+        ## テスト
+        
+        Note that テスト consists of Non-ASCII characters of E3 83 86 E3 82 B9 E3 83 88 in UTF-8.
```

Comment:
very minor: Rather than this content in the markup, this information could be more useful near the `heading.anchor` assertion. 
```swift
XCTAssertEqual(heading.anchor, "%E3%83%86%E3%82%B9%E3%83%88", "The UTF-8 representation of テスト is E3 83 86 E3 82 B9 E3 83 88")
```

If could also help explain the test assertion if there was a second assertion that verified that it matched the percent escaped value:
```swift
XCTAssertEqual(heading.anchor, "テスト".addingPercentEncoding(withAllowedCharacters: .urlFragmentAllowed))
```

That way someone who reads this test can understand that `%E3%83%86%E3%82%B9%E3%83%88` is the percent encoded fragment.